internal/cmd/versions.go

```diff
@@ -0,0 +1,69 @@
+package cmd
+
+import (
+	"fmt"
+	"runtime/debug"
+	"strings"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// envOutput determines whether to output as environment settings
+	envOutput bool
+)
+
+// getVersionsCommand returns the server cobra command to be executed.
+func getVersionsCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "versions",
+		Aliases: []string{"version"},
+		Short:   "Show versions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return versions()
+		},
+	}
+
+	cmd.PersistentFlags().BoolVarP(&envOutput, "env", "e", false,
+		"If set, output as environment variable settings.")
+
+	return cmd
+}
+
+// versions shows the versions of the Envoy Gateway.
+func versions() error {
+	envoyVersion := strings.Split(ir.DefaultProxyImage, ":")[1]
+
+	if envOutput {
+		fmt.Printf("ENVOY_VERSION=\"%s\"\n", envoyVersion)
+	} else {
+		fmt.Printf("Envoy:       %s\n", envoyVersion)
+	}
+
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		return fmt.Errorf("could not read build info")
+	}
+
+	foundGatewayAPI := false
+
+	for _, dep := range bi.Deps {
+		if dep.Path == "sigs.k8s.io/gateway-api" {
+			if envOutput {
+				fmt.Printf("GATEWAYAPI_VERSION=\"%s\"\n", dep.Version)
+			} else {
+				fmt.Printf("Gateway API: %s\n", dep.Version)
+			}
+
+			foundGatewayAPI = true
+			break
+		}
+	}
+
+	if !foundGatewayAPI {
+		return fmt.Errorf("could not find Gateway API version")
+	}
```

Comment:
It's kinda weird to me that it reports the Envoy version and the Gateway API version, but not its own version.

If this isn't intended to be a user-facing command, and just used internally to build the docs, then I guess that's fine.

In `ocibuild` what I did for such commands that were only used to build the docs is to put them behind a `//go:build aux` comment, so that I could `go run -tags=aux` them, but they wouldn't be included in the compiled binary. https://github.com/datawire/ocibuild/blob/master/main_aux.go

Reply:
If this _is_ meant to be a user-facing thing, then it should probably use Cobra's built-in version stuff; set `cmd.Version` on the root command, and adjust `cmd.SetVersionTemplate()` as necessary.

Reply:
@LukeShu We discussed this at the Envoy Gateway meeting today, and landed on the side of "it should be user-visible". You OK approving this version and letting a subsequent PR switch to the Cobra stuff? I'd like to kick around the best way to do that with you.

Reply:
Sure.